### PR TITLE
refactor: [DHIS2-17750] replace material ui Card for Widget

### PR DESCRIPTION
**Pull request by simonadomnisoru**
[DHIS2-17750](https://dhis2.atlassian.net/browse/DHIS2-17750)

**Tech summary**
- The material-ui `Card` was still used to display the program rules effects when adding a new event to a single event program. I replaced the Card for the Widgets

[DHIS2-17750]: https://dhis2.atlassian.net/browse/DHIS2-17750?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by eirikhaugstulen — COMMENTED** [review]
Hey Simona! The updated version looks like a big improvement from the old cards and it will make our event pages seem much more modern.

Left some small comments on the PR itself, and then I have two questions that may or may not be relevant to this issue, but I'll post it here either way. Feel free to disregard if it should be separate tickets.

1. Is this PR not relevant for the registration page? It looks like create new in program still uses the old widgets.

image


2. Do you think we could change the styling so that the right column is not offset like it currently is?

image
As you can see, the left and right column does not start at the same height, and I'm not sure why.


I think something like this would be an improvement:
image

All I did was to remove the `marginTop: 10` from the mapping in `withDataEntryOutput` and in `DataEntry.component.js`, and then changed `StickyOnScroll > stickyContainerAbsolute` to something like this: 
```css
stickyContainerAbsolute: {
    position: 'static',
    '& > div > *:not(:first-child)': {
        marginTop: '10px',
    },
},
```

Let me know what you think. Thank you 🎉

File: src/core_modules/capture-core/components/DataEntry/dataEntryOutput/withIndicatorOutput.js
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/DataEntry/dataEntryOutput/withIndicatorOutput.js:1]: Good catch! I added back the functionality

File: src/core_modules/capture-core/components/DataEntry/dataEntryOutput/withFeedbackOutput.js [hunk 1]
```diff
@@ -1,98 +1,31 @@
...
-                            {i18n.t('Feedback')}
-                            <Menu dense>
-                                {feedbackItems && this.renderFeedbackItems(feedbackItems, classes)}
-                            </Menu>
-                        </Card>
-                    }
-                </div>
-            );
+            const feedback = this.getItems();
+            return <WidgetFeedback feedback={feedback} emptyText={i18n.t('No feedback')} />;
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/DataEntry/dataEntryOutput/withFeedbackOutput.js]: Yes, I implemented the same pattern now with the latest commit.

**Comment by simonadomnisoru** [comment]
> Hey Simona! The updated version looks like a big improvement from the old cards and it will make our event pages seem much more modern.
> 
> Left some small comments on the PR itself, and then I have two questions that may or may not be relevant to this issue, but I'll post it here either way. Feel free to disregard if it should be separate tickets.
> 
> 1. Is this PR not relevant for the registration page? It looks like create new in program still uses the old widgets.
> 2. Do you think we could change the styling so that the right column is not offset like it currently is?

Hey @Eirik,
Regarding the widgets on the TEI registration page, the Material UI components are not used on that page, so this ticket is not relevant there. I implemented the rest of your suggestions.
Thank you for the feedback!

**Review by eirikhaugstulen — APPROVED** [review]
This is a major upgrade @simonadomnisoru! LGTM!

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.42 version

**Pull request merged**